pygmt/src/shift_origin.py

```diff
@@ -1,33 +1,57 @@
 """
 shift_origin - Shift plot origin in x and/or y directions.
 """
+from __future__ import annotations
 
 from pygmt.clib import Session
 
 
-def shift_origin(self, xshift=None, yshift=None):
-    """
+def shift_origin(
+    self, xshift: float | str | None = None, yshift: float | str | None = None
+) -> None:
```

Comment:
Do we plan adding the `Return type` section for all methods, even it is `None`?

![shiftorigin_returntype](https://github.com/GenericMappingTools/pygmt/assets/94163266/a1a62bcd-acc4-4678-a311-d3a5b74adc55)

Reply:
I was also debating about this. I agree that the `None` return type makes little sense so should be removed.